### PR TITLE
Update description to point to jaegertracing.io docs, and remove the …

### PR DESCRIPTION
**Pull request by objectiser**
…example as atleast one is made available via the alm-examples via the operatorhub.io website

Signed-off-by: Gary Brown <gary@brownuk.com>

### CODE REVIEW

File: deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -94,19 +94,19 @@ spec:
     * Service dependency analysis
     * Performance / latency optimization
     * OpenTracing compatible data model
-    * Multiple storage backends: Cassandra, Elasticsearch, memory.
+    * Multiple storage backends: Cassandra, Elasticsearch, Memory, Badger.
```
Comment (JStickler) [inline deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml]: My former editor used to say "Alphabetize unless there's a reason not to."  Unless these are in order of preference, we should probably update to read "Badger, Cassandra, Elasticsearch, Memory."

File: deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -94,19 +94,19 @@ spec:
...
 
     ### Operator features
 
-    * **Multiple modes** - Supports `allInOne`, `production`, and `streaming` [modes of deployment](https://github.com/jaegertracing/jaeger-operator#strategies).
+    * **Multiple modes** - Supports `allInOne`, `production`, and `streaming` [modes of deployment](https://www.jaegertracing.io/docs/latest/operator/#deployment-strategies).
```
Comment (JStickler) [inline deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml]: Are the "modes" here deployment strategies or Helm charts or something else?  (Wondering if "production" should be "production-elasticsearch"? And if that is "allInOne" or "all-in-one"?)

File: deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -94,19 +94,19 @@ spec:
...
 
-    * **Configuration** - Directly pass down all supported Jaeger configuration through the Operator.
+    * **Configuration** - Directly pass down all supported [Jaeger configuration](https://www.jaegertracing.io/docs/latest/operator/#configuring-the-custom-resource) through the Operator.
```
Comment (JStickler) [inline deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml]: I'm not sure what we mean by "pass down" here?  I think what we want to say is something more like "The Operator manages configuration information when installing Jaeger instances."

File: deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml (hunk 2 above)
Comment (objectiser) [reply deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml]: It is actually the 'strategy' (architecture) used - independent of the storage which is specified separately.
It is `allInOne`.
This is different to the template used in the maistra operator's CR, which is a combination of the strategy and storage choice.

File: deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml (hunk 1 above)
Comment (objectiser) [reply deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml]: Will update.

File: deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml (hunk 3 above)
Comment (objectiser) [reply deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml]: Yep sounds better. Thanks

File: deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml (hunk 2 above)
Comment (JStickler) [reply deploy/olm-catalog/jaeger-operator/1.14.0/jaeger-operator.v1.14.0.clusterserviceversion.yaml]: Different upstream/downstream might explain why I keep getting confused!

**Pull request merged**